### PR TITLE
camerad: OS HDR

### PR DESCRIPTION
**Pull request by ZwX1616**

### CODE REVIEW

**Review by adeebshihadeh — CHANGES_REQUESTED** [review]
The debayer kernel is getting super messy. It's really important that as a project we strictly improve. Some general things that would improve the kernel a lot:
* less magic numbers
* more descriptive names and comments
* more arguments instead of ifdefs, e.g. define the color correction values in `SensorInfo` and pass them in

File: system/camerad/sensors/sensor.h [hunk 1]
```diff
@@ -28,6 +28,8 @@ class SensorInfo {
   uint32_t extra_height = 0;
   int registers_offset = -1;
   int stats_offset = -1;
+  int hdr_offset = -1;
+  bool bggr;
```
[the remark this answers is not in the document]
Comment (ZwX1616) [reply system/camerad/sensors/sensor.h:32]: well there'll only be two really, like odd and even, at least in our case

File: system/camerad/cameras/camera_common.h [hunk 2]
```diff
@@ -24,6 +24,13 @@ const bool env_debug_frames = getenv("DEBUG_FRAMES") != NULL;
 const bool env_log_raw_frames = getenv("LOG_RAW_FRAMES") != NULL;
 const bool env_ctrl_exp_from_params = getenv("CTRL_EXP_FROM_PARAMS") != NULL;
 
+typedef struct AutoExposureRect {
```
Comment (adeebshihadeh) [inline system/camerad/cameras/camera_common.h:27]: this is just a "rect". move to common/util.h?

**Review by adeebshihadeh — CHANGES_REQUESTED** [review]
Looks a lot better now but this is still pretty massive. It's OS HDR + what should be a no-op refactor. Let's split those two up. Also, I see the test is deleted; how did you verify that it's no-op on AR/OX?

**Pull request merged**